rasa/core/actions/action.py

```diff
@@ -1234,7 +1234,7 @@ async def run(
                     if not isinstance(slot, ListSlot):
                         value = value[-1]
 
-                    if tracker.get_slot(slot.name) != value:
+                    if value is not None or slot.value is not None:
```

Comment:
I don't remember all the unit test cases, but I'd add this test if it doesn't exist, where the tracker slot has a value, but the new value is `None`.